versions.mk

```diff
@@ -28,7 +28,7 @@ NVIDIA_DOCKER_VERSION := 2.14.0
 NVIDIA_CONTAINER_RUNTIME_VERSION := 3.14.0
 
 CUDA_VERSION := 12.3.1
-GOLANG_VERSION := 1.20.5
+GOLANG_VERSION := 1.21.5
```

Comment:
Let me know if we are okay to update this. I am assuming we might run into the glibc version skew here as well

Reply:
Yes, we would have to see whether these work in older distributions. I already have issues with `1.20.5` on some systems.

Reply:
I can revert it then. We definitely need to think of a solution for this though given 1.20 is close to EOL. Are we dependent on the glibc that comes from the `cuda-base-image`?

Reply:
We don't build in the CUDA base images. We build in `ubuntu:18.04` and `centos:7` (`centos:8` for ppc64le).

Let's drop the golang bump and do that as a follow-up.